playbooks/roles/ocp-lso/tasks/main.yaml

```diff
@@ -0,0 +1,236 @@
+
+    # Wait for LocalVolumeSet to be created
+    - name: Wait for LocalVolumeSet to be created
+      command: oc get localvolumeset localblock -n "{{ lso_namespace }}"
+      register: lvs_status
+      retries: 5
+      delay: 10
+      until: lvs_status.rc == 0
+      failed_when: lvs_status.rc != 0  # Fail if LocalVolumeSet is not created within 5 retries
+
+    - name: Verify creation of diskmanager pods
+      command: oc get pods -n "{{ lso_namespace }}" --selector app=diskmaker-manager
+      register: check_pods
+      failed_when: check_pods.rc != 0
+
+    - name: Verify the creation of StorageClass
+      command: oc get sc localblock
+      register: check_sc
+      failed_when: check_sc.rc != 0
+
+    - name: Wait for persistent volumes to be created
+      pause:
+        minutes: 1
+
+    - name: Verify the status of Persistent Volumes
+      command: oc get pv --selector storage.openshift.com/owner-kind=LocalVolumeSet
+      register: check_pv
+      failed_when: check_pv.rc != 0
+
+    - name: Delete existing PersistentVolumeClaim and Pod (if they exist)
+      kubernetes.core.k8s:
+        state: absent
+        definition:
+          - kind: PersistentVolumeClaim
+            apiVersion: v1
+            metadata:
+              name: test-local-claim
+          - apiVersion: v1
+            kind: Pod
+            metadata:
+              name: busybox-lso
+              namespace: default
+      register: delete_pvc
+      failed_when: false  # Do not fail if delete command is not successful (e.g., resource doesn't exist)
+      changed_when: delete_pvc is changed  # Mark as changed only if a deletion actually occurred
+
+    # Testing localblock storage class
+    - name: Create PersistentVolumeClaim YAML
+      kubernetes.core.k8s:
+        definition:
+          - kind: PersistentVolumeClaim
+            apiVersion: v1
+            metadata:
+              name: test-local-claim
+              namespace: default
+            spec:
+              accessModes:
+              - ReadWriteOnce
+              volumeMode: Block
+              resources:
+                requests:
+                  storage: 100Gi
+              storageClassName: localblock
+
+          - apiVersion: v1
+            kind: Pod
+            metadata:
+              name: busybox-lso
+              namespace: default
+            spec:
+              volumes:
+                - name: local-data
+                  persistentVolumeClaim:
+                    claimName: test-local-claim
+              containers:
+                - name: busybox
+                  image: quay.io/powercloud/busybox:ubi
+                  command: ['sh', '-c', 'echo "Hello, Kubernetes!" && sleep 600']
+                  volumeDevices:
+                    - devicePath: "/dev/lso"
+                      name: local-data
+
+    - name: Verify PVC status
+      command: oc get pvc
+      register: pvc_status
+      failed_when: pvc_status.rc != 0
```

Comment:
Please validate PVC based name, namespace and its state.

Reply:
done